share/functions/help.fish

```diff
@@ -62,11 +62,16 @@ function help --description 'Show help for the fish shell'
                 set fish_browser xdg-open
             end
 
-            # On OS X, we go through osascript by default
+            # On OS X, we use osascript < 10.12.5, and open after (see #4035)
             if test (uname) = Darwin
-                if type -q osascript
+	        set -l version (sw_vers -productVersion | string split .)
+		if [ $version[1] -gt 10 ]
+		    or [ $version[1] -eq 10 -a $version[2] -gt 12 ]
+		    or [ $version[1] -eq 10 -a $version[2] -eq 12 -a $version[3] -ge 5 ]
```

Comment:
Is this guaranteed to always have 3 elements?

This would currently error (with a message) if it doesn't. I'd probably go with just unconditionally appending three "0" elements.